sdk/network/mgmt/src/main/java/com/azure/management/network/implementation/ApplicationGatewayImpl.java

```diff
@@ -578,30 +578,30 @@ private static ApplicationGatewayFrontendImpl useSubnetFromIPConfigForFrontend(
      * @param byName object found by name
      * @param byPort object found by port
      * @param name the desired name of the object
-     * @return true if already found, false if ok to create, null if conflict
+     * @return positive if already found, 0 if ok to create, negative if conflict
      */
-    <T> Boolean needToCreate(T byName, T byPort, String name) {
+    <T> int needToCreate(T byName, T byPort, String name) {
```

Comment:
Could be reviewed carefully.

Reply:
Em, what does spotbugs say on Boolean?

Reply:
If return null for Boolean, it will cause NPE for boolean as return type.

A method that returns either Boolean.TRUE, Boolean.FALSE or null is an accident waiting to happen. This method can be invoked as though it returned a value of type boolean, and the compiler will insert automatic unboxing of the Boolean value. If a null value is returned, this will result in a NullPointerException.

https://github.com/codacy/codacy-findbugs/blob/master/src/main/resources/docs/description/NP_BOOLEAN_RETURN_NULL.md

Reply:
Agree. But `int` does not make it clearer as well (at least for human). Who would know a integer only can take 3 values? If you do like clarity, do enum.

Reply:
done

Reply:
done